### ISSUE TITLE
[Bug]: jsx-newline still considering comments with { "prevent": true, "allowMultilines": true }

### ISSUE THREAD

**Issue by bryanhaney-shipengine** [jsx-eslint/eslint-plugin-react#3526]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

Per https://github.com/jsx-eslint/eslint-plugin-react/issues/2926, this has been reported as fixed, but I am still seeing `Multiline JSX elements should start in a new line` when a comment is above a multiline JSX block.

- config:
`"react/jsx-newline": ["warn", { "prevent": true, "allowMultilines": true }],`

- example of invalid code:
```tsx
{/* fake-eslint-disable-next-line react/forbid-component-props */}
<div>
  <p>Cool Beans</p>
  <p>Sweet</p>
</div>
```

Changing the config to `"prevent": false` works as expected (it does not consider the comment)

### Expected Behavior

Comments above multiline JSX blocks should be ignored if jsx-newline is enabled with config options: `{ "prevent": true, "allowMultilines": true }`

I expect this code to pass:

```tsx
{/* fake-eslint-disable-next-line react/forbid-component-props */}
<div>
  <p>Cool Beans</p>
  <p>Sweet</p>
</div>
```

### eslint-plugin-react version

v7.32.1

### eslint version

v8.28.0

### node version

v17.9.0
